api/src/main/java/org/apache/iceberg/ExpireSnapshots.java

```diff
@@ -82,4 +82,16 @@
    * @return this for method chaining
    */
   ExpireSnapshots deleteWith(Consumer<String> deleteFunc);
+
+  /**
+   * Allows expiration of snapshots without any cleanup of underlying manifest or data files.
+   * <p>
+   * Allows control in removing data and manifest files which may be more efficiently removed using
+   * a distributed framework through the actions API.
+   * </p>
+   *
+   * @param clean setting this to false will skip deleting expired manifests and files
+   * @return this for method chaining
+   */
+  ExpireSnapshots deleteExpiredFiles(boolean clean);
```

Comment:
Should this be `cleanExpiredFiles` instead? I think that makes more sense because this actually avoids the work of finding the files that have expired, not calling delete for them. And that's what you call it elsewhere, so it makes sense to me.

Reply:
sgtm